### PR TITLE
Added threshold for minimum backward transfer.

### PR DESCRIPTION
**Pull request by VolodymyrBabaiev**
This PR introduced a minimum coin amount for withdrawal requests. Withdrawal transactions with a lower amount of coins will be rejected.

### CODE REVIEW

**Review by i-Alex — CHANGES_REQUESTED** [review]
Good!

File: sdk/src/main/java/com/horizen/transaction/SidechainTransaction.java [hunk 1]
```diff
@@ -86,6 +87,11 @@ public final void semanticValidity() throws TransactionSemanticValidityException
             }
             // check coins box value
             if(box instanceof CoinsBox || box instanceof WithdrawalRequestBox) {
+                if (box instanceof WithdrawalRequestBox
+                        && box.value() < withdravalThreshold) {
+                    throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                            "Withdrawal request box [%d] value is belove the threshhold[%d].", id(), i, withdravalThreshold));
```
[the remark this answers is not in the document]
Comment (i-Alex) [reply sdk/src/main/java/com/horizen/transaction/SidechainTransaction.java]: actual

File: qa/sc_bwt_minimum_value.py [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+        assert_equal(0, we0_sc_cert["epochNumber"], "Sidechain epoch number in certificate is wrong.")
+        assert_equal(we0_end_epoch_cum_sc_tx_comm_tree_root, we0_sc_cert["endCumulativeScTxCommitmentTreeRoot"],
+                     "Sidechain endEpochCumScTxCommTreeRoot in certificate is wrong.")
+        assert_equal(0, len(we0_sc_cert["backwardTransferOutputs"]), "Backward transfer amount in certificate is wrong.")
+        assert_equal(we0_certHash, we0_sc_cert["hash"], "Certificate hash is different to the one in MC.")
+
+
+        # Checking withdrawCoins
+        # Try to withdraw coins from SC to MC: amount below the dust threshold
+        mc_address1 = self.nodes[0].getnewaddress()
```
Comment (i-Alex) [inline qa/sc_bwt_minimum_value.py:177]: Test flow until this point is a duplication of `sc_backward_transfer.py`
What is the sense to test Cert with no BTs here?
Please, remove the redundant part and keep only the cases of BT with minimum amount.
Also, update the test description.

File: sdk/src/main/java/com/horizen/transaction/SidechainTransaction.java [hunk 3]
```diff
@@ -86,6 +87,11 @@ public final void semanticValidity() throws TransactionSemanticValidityException
             }
             // check coins box value
             if(box instanceof CoinsBox || box instanceof WithdrawalRequestBox) {
+                if (box instanceof WithdrawalRequestBox
+                        && box.value() < withdrawalThreshold) {
+                    throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                            "Withdrawal request box [%d] value is belove the threshhold[%d].", id(), i, withdrawalThreshold));
```
Comment (i-Alex) [inline sdk/src/main/java/com/horizen/transaction/SidechainTransaction.java:93]: belove -> below, threshhold =-> threshold

File: sdk/src/main/java/com/horizen/utils/ZenCoinsUtils.java [hunk 4]
```diff
@@ -5,6 +5,20 @@ private ZenCoinsUtils() {}
 
     public static long COIN = 100000000;
     public static long MAX_MONEY = 21000000 * COIN;
+    public static long MC_DEFAULT_FEE_RATE = 100; // Default fee rate set in Mainchain.
+                                                  // Value stored in satoshi per kilobyte.
+
+    /**
+      * This function calculates the minimum value of Backward Transfer Transaction that Mainchain would accept.
```
[the remark this answers is not in the document]
Comment (i-Alex) [reply sdk/src/main/java/com/horizen/utils/ZenCoinsUtils.java:12]: still actual

**Review by i-Alex — CHANGES_REQUESTED** [review]

File: qa/sc_bwt_minimum_value.py (hunk 2 above)
Comment (VolodymyrBabaiev) [reply qa/sc_bwt_minimum_value.py:105]: Done.

**Review by i-Alex — APPROVED** [review]

**Review by albertog78 — APPROVED** [review]

**Pull request merged**